src/iden3comm/packers/jws.ts

```diff
@@ -0,0 +1,174 @@
+import { extractPublicKeyBytes, getDIDComponentById, resolveDIDDocument } from '../utils/did';
+import { keyPath, KMS } from '../../kms/';
+
+import { Signer, verifyJWS } from 'did-jwt';
+import { Resolvable, VerificationMethod, parse } from 'did-resolver';
+import {
+  byteDecoder,
+  byteEncoder,
+  bytesToHex,
+  decodeBase64url,
+  encodeBase64url
+} from '../../utils';
+export type SignerFn = (vm: VerificationMethod, data: Uint8Array) => Signer;
+
+/**
+ * Packer that can pack message to JWZ token,
+ * and unpack and validate JWZ envelope
+ * @exports
+ * @beta
+ * @class ZKPPacker
+ * @implements implements IPacker interface
+ */
+export class JWSPacker implements IPacker {
+  // readonly vmPubkeyExtractorHandlerMap = {
+  //   ES256K: getPubKeyHexFromVm,
+  //   'ES256K-R': getPubKeyHexFromVm
+  // };
+
+  // readonly verifySignatureHandlerMap = {
+  //   ES256K: verifySignatureSecp256K1,
+  //   'ES256K-R': getPubKeyHexFromVm
+  // };
+
+  constructor(
+    private readonly _kms: KMS,
+    private readonly _documentResolver: Resolvable = { resolve: resolveDIDDocument }
+  ) {}
+  /**
+   * creates JSON Web Signature token
+   *
+   * @param {Uint8Array} payload - serialized message
+   * @param {PackerParams} params - sender id and proving alg are required
+   * @returns `Promise<Uint8Array>`
+   */
+  async pack(
+    payload: Uint8Array,
+    params: PackerParams & {
+      alg: string;
+      did: string;
+      kid?: string;
+      signer?: SignerFn;
+    }
+  ): Promise<Uint8Array> {
+    if (!params.alg) {
+      throw new Error('Missing algorithm');
+    }
+    const message = JSON.parse(byteDecoder.decode(payload));
+
+    const from = message.from ?? '';
+    if (params.did !== message.from) {
+      throw new Error('DID in params does not match DID in message');
+    }
+
+    if (!from) {
+      throw new Error('Missing DID');
+    }
+
+    const vmTypes: string[] = SUPPORTED_PUBLIC_KEY_TYPES[params.alg];
+    if (!vmTypes?.length) {
+      throw new Error(`No supported signature types for algorithm ${params.alg}`);
+    }
+
+    const { didDocument } = await this._documentResolver.resolve(from);
+
+    const section = 'authentication';
+    const authSection = didDocument[section] ?? [];
+
+    const vms = authSection
+      .map((key) =>
+        typeof key === 'string'
+          ? getDIDComponentById(didDocument, key ?? didDocument.id, section)
+          : key
+      )
+      .filter(Boolean);
+
+    if (!vms.length) {
+      throw new Error(`No keys found in ${section} section of DID document ${didDocument.id}`);
+    }
+
+    // try to find a managed signing key that matches keyRef
+    const vm = params.kid ? vms.find((vm) => vm.id === params.kid) : vms[0];
+
+    // || vm.blockchainAccountId === params.kid
+
+    if (!vm) {
+      throw new Error(
+        `No key found with id ${params.kid} in ${section} section of DID document ${didDocument.id}`
+      );
+    }
+
+    if (vm.blockchainAccountId && !params.signer) {
```

Comment:
if !vm.publicKeyHex &&  vm.ethereumAddress || vm.blockchainAccounId

Reply:
maybe firstly we can do       const { publicKeyBytes, kmsKeyType } = extractPublicKeyBytes(vm); 
and if we don't have a hex -check this condition